tests/run_examples.py

```diff
@@ -0,0 +1,304 @@
+    logging.basicConfig(format="%(message)s", level=logging.INFO, stream=sys.stdout)
+    logger = logging.getLogger('galsim')
+    config = galsim.config.ReadConfig('json/demo9.json', logger=logger)[0]
+    print('Running demo9.json')
+    new_params = { 'output.skip' : { 'type' : 'List', 'items' : [0,0,0,0,0,1] } }
+    galsim.config.Process(config, logger=logger, new_params=new_params, njobs=3, job=1)
+    galsim.config.Process(config, logger=logger, new_params=new_params, njobs=3, job=2)
+    galsim.config.Process(config, logger=logger, new_params=new_params, njobs=3, job=3)
+    new_params = { 'output.noclobber' : True }
+    galsim.config.Process(config, logger=logger, new_params=new_params)
+    for dir_num in range(1,5):
+        for file_num in range(5):
+            file_name = 'nfw%d/cluster%04d.fits'%(dir_num, file_num)
+            truth_name = 'nfw%d/truth%04d.dat'%(dir_num, file_num)
+            assert check_diff.same('output/'+file_name , 'output_json/'+file_name)
+            assert check_diff.same('output/'+truth_name , 'output_json/'+truth_name)
+
+@timer
+@in_examples
+def test_demo10():
+    """Check that demo10 makes the same image using demo10.py and demo10.yaml.
+    """
+    import demo10
+    import check_diff
+    print('Running demo10.py')
+    demo10.main([])
+    logging.basicConfig(format="%(message)s", level=logging.INFO, stream=sys.stdout)
+    logger = logging.getLogger('galsim')
+    config = galsim.config.ReadConfig('demo10.yaml', logger=logger)[0]
+    print('Running demo10.yaml')
+    galsim.config.Process(config, logger=logger)
+    assert check_diff.same('output/power_spectrum.fits', 'output_yaml/power_spectrum.fits')
+
+@timer
+@in_examples
+def test_demo11():
+    """Check that demo11 makes the same image using demo11.py and demo11.yaml.
+    """
+    import demo11
+    import check_diff
+    print('Running demo11.py')
+    demo11.main([])
+    logging.basicConfig(format="%(message)s", level=logging.INFO, stream=sys.stdout)
+    logger = logging.getLogger('galsim')
+    config = galsim.config.ReadConfig('demo11.yaml', logger=logger)[0]
+    print('Running demo11.yaml')
+    galsim.config.Process(config, logger=logger)
+    assert check_diff.same('output/tabulated_power_spectrum.fits.fz',
+                           'output_yaml/tabulated_power_spectrum.fits.fz')
+
+@timer
+@in_examples
+def test_demo12():
+    """Check that demo12 runs properly.
+    """
+    import demo12
+    import check_diff
+    print('Running demo12.py')
+    demo12.main([])
+    # There is no demo12.yaml yet, so all this does is check for syntax errors in demo12.py.
+
+@timer
+@in_examples
+def test_demo13():
+    """Check that demo13 runs properly.
+    """
+    import demo13
+    import check_diff
+    print('Running demo13.py')
+    demo13.main([])
+    # There is no demo13.yaml yet, so all this does is check for syntax errors in demo13.py.
+
+
+if __name__ == "__main__":
+    shutil.rmtree('../examples/output')
```

Comment:
Should probably make this path with respect to `__file__`, in case someone starts this script from a directory other than $GALSIM_DIR/tests/.

Also, I think this line crashes if the target directory doesn't exist.